### PR TITLE
implement storage migration for ceremonies v2 Reputation type

### PR DESCRIPTION
**Pull request by brenzi**
remainder of #373

### CODE REVIEW

File: ceremonies/src/migrations.rs [hunk 1]
```diff
@@ -220,4 +328,34 @@ mod test {
...
+			let state = v2::MigrateToV2::<TestRuntime>::pre_upgrade().unwrap();
+			let _weight = v2::MigrateToV2::<TestRuntime>::on_runtime_upgrade();
+			v2::MigrateToV2::<TestRuntime>::post_upgrade(state).unwrap();
+
+			// Check that all values got migrated.
+
+			assert_eq!(
+				crate::ParticipantReputation::<TestRuntime>::get((cid, 1), alice),
+				Reputation::VerifiedLinked(1)
+			);
```
Comment (clangenb) [inline ceremonies/src/migrations.rs:358]: Mean the code is very simple, but we might want to test that the other reputation variants are unaltered.

File: ceremonies/src/migrations.rs [hunk 2]
```diff
@@ -22,6 +22,52 @@ mod v0 {
...
+					encointer_primitives::ceremonies::Reputation::UnverifiedReputable
+				},
+				Reputation::VerifiedLinked => {
+					encointer_primitives::ceremonies::Reputation::VerifiedLinked(linked_cindex)
+				},
+				Reputation::VerifiedUnlinked => {
+					encointer_primitives::ceremonies::Reputation::VerifiedUnlinked
+				},
+				Reputation::Unverified => encointer_primitives::ceremonies::Reputation::Unverified,
+			}
```
Comment (clangenb) [inline ceremonies/src/migrations.rs:56]: Not really necessary to change, but still more readable and less error-prone:


```rust
			match self {
				Reputation::VerifiedLinked => {
					encointer_primitives::ceremonies::Reputation::VerifiedLinked(linked_cindex)
				},
				reputation => reputation
			}
```

**Review by clangenb — COMMENTED** [review]
Looks good to me, some tiny comments only.

File: ceremonies/src/migrations.rs (hunk 1 above)
Comment (brenzi) [reply ceremonies/src/migrations.rs:358]: hmmm, can't we trust the compiler with that?
I get your point about test coverage, but in this case I think the compiler is restrictive enough

File: ceremonies/src/migrations.rs (hunk 2 above)
Comment (brenzi) [reply ceremonies/src/migrations.rs:56]: wasn't aware of that. I'm surprised that this is considered type-safe by rustc

Comment (brenzi) [reply ceremonies/src/migrations.rs:56]: I was right, it isn't ;-)

File: ceremonies/src/migrations.rs (hunk 1 above)
Comment (clangenb) [reply ceremonies/src/migrations.rs:358]: only if you correctly implemented the match statement ;)

File: ceremonies/src/migrations.rs (hunk 2 above)
Comment (brenzi) [reply ceremonies/src/migrations.rs:56]: `expected `encointer_primitives::ceremonies::Reputation`, found `migrations::v1::Reputation``

Comment (clangenb) [reply ceremonies/src/migrations.rs:56]: well, enum variants are all the same type.

**Pull request merged**